src/test/java/com/fasterxml/jackson/databind/cfg/CacheProviderTest.java

```diff
@@ -160,21 +182,84 @@ public void testDefaultCacheProviderSharesCache() throws Exception
         ObjectMapper mapper2 = JsonMapper.builder()
                 .cacheProvider(cacheProvider)
                 .build();
-        
+
         // Act 
         // 3. Add two different types to each mapper cache
         mapper1.readValue("{\"point\":24}", RandomBean.class);
         mapper2.readValue("{\"height\":24}", AnotherBean.class);
-        
+
         // Assert
         // 4. Should have created two cache instance
         assertEquals(2, cacheProvider.createCacheCount());
     }
 
     @Test
-    public void testBuilderBuildWithDefaults() throws Exception
+    public void testBuilderValueValidation() throws Exception
     {
         // does not throw
-        DefaultCacheProvider.builder().build();
+        DefaultCacheProvider.builder()
+                .build();
+        DefaultCacheProvider.builder()
+                .maxDeserializerCacheSize(Integer.MAX_VALUE)
+                .maxSerializerCacheSize(Integer.MAX_VALUE)
+                .build();
+
+        // validation fails
+        _verifyValidationFail(() -> DefaultCacheProvider.builder().maxDeserializerCacheSize(-1),
+                "Cannot set maxDeserializerCacheSize to a negative value");
+        _verifyValidationFail(() -> DefaultCacheProvider.builder().maxSerializerCacheSize(-1),
+                "Cannot set maxSerializerCacheSize to a negative value");
+    }
+
+    private void _verifyValidationFail(Runnable cacheProviderBuilder, String expectedErrorMsg) {
+        try {
+            cacheProviderBuilder.run();
+            fail("Should not reach here");
+        } catch (IllegalArgumentException e) {
+            assertTrue(e.getMessage().contains(expectedErrorMsg));
+        }
+    }
+
+    @Test
+    public void testDefaultCacheProviderConfigSerializerCache() throws Exception
+    {
+        CacheProvider cacheProvider = DefaultCacheProvider.builder()
+                .maxSerializerCacheSize(1234)
+                .build();
+        ObjectMapper mapper = JsonMapper.builder()
+                .cacheProvider(cacheProvider)
+                .build();
+
+        String json = mapper.writeValueAsString(new SerBean());
+
+        assertEquals("{\"slide\":123}", json);
+    }
+
+    @Test
+    public void testDefaultCacheProviderConfigSerializerCacheSizeZero() throws Exception
```

Comment:
Similarly, what exactly is being tested?

Reply:
Minor refactoring to reduce duplicated code would be useful in that case.
Although also checking that cache being constructed specified size would be useful -- but wouldn't want to add methods just for test cases.

So yeah, reducing code duplication would be good, I'm ok with the tests otherwise.